segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ReadOperation.java

```diff
@@ -111,6 +114,8 @@
                                         // Now read.
                                         return readData(txn);
                                     }, chunkedSegmentStorage.getExecutor())
+                                    // HACK: delay a while after read chunk
+                                    .thenCompose(vv -> createDelayFuture(hackDelayMillis))
```

Comment:
```.thenComposeAsync(..., , chunkedSegmentStorage.getExecutor())```
We don't want any storage code executed on ForkJoin thread pool. (dependent actions run on same threadpool that completed the future, so by mistake whole LTS might end up running on ForkJoin bringing down the house)

Reply:
I am not using thenComposeAsync without an executor supplier, but a sync thenCompose which is executed in the same thread of the previous stage, no thread transfer here.

Reply:
@realAaronWu I think that this assumption is not true, based on the documentation: https://docs.oracle.com/javase/8/docs/api/?java/util/concurrent/CompletableFuture.html

> Actions supplied for dependent completions of non-async methods may be performed by the thread that completes the current CompletableFuture, or by any other caller of a completion method.

If I understand correctly, it is not guaranteed that `createDelayFuture()` is going to be executed by the same thread as the previous stage (it could be). But it could be also executed on the thread running the `call()` function.

Also, I don't think this is going to be run in the default `ForkJoinPool`, because according to the docs for doing this we should use `thenComposeAsync()` without any executor supplied:
> All async methods without an explicit Executor argument are performed using the ForkJoinPool.commonPool()

Reply:
@RaulGracia  it does not matter which thread is executing this, as long as it is in the chunkedSegmentStorage thread pool.   The only possibility let other threads to execute is that the completableFuture is exposed by mistake outside, and we explicitly complete this completableFuture  in the code, then the next stage will be transfered to the very thread completing the previous one.

Reply:
The reason I use the sync compose it to minimize the influnce of the delay in the threads pool. The very thread which is completing the previous read/write chunk stage is the correct one to suffer this delay.